include/RED4ext/Containers/Span.hpp

```diff
@@ -0,0 +1,232 @@
+#pragma once
+
+#include <algorithm>
+#include <cassert>
+#include <cstdint>
+#include <stdexcept>
+
+#include <RED4ext/Common.hpp>
+#include <RED4ext/Detail/Containers/ArrayIterator.hpp>
+
+namespace RED4ext
+{
+template<typename T>
+struct Span
+{
+    using ValueType = T;
+    using Reference = ValueType&;
+    using ConstReference = const Reference;
+    using Pointer = ValueType*;
+    using ConstPointer = const Pointer;
+
+    using SizeType = std::uint32_t;
+    using DifferenceType = std::ptrdiff_t;
+
+    using Iterator = Detail::ArrayIterator<ValueType, Span>;
+    using ConstIterator = Detail::ArrayIterator<const ValueType, Span>;
+    using ReverseIterator = std::reverse_iterator<Iterator>;
+    using ConstReverseIterator = std::reverse_iterator<ConstIterator>;
+
+    Span()
+        : beginPtr(nullptr)
+        , endPtr(nullptr)
+    {
+    }
+
+    Span(Pointer aBegin, Pointer aEnd)
+        : beginPtr(aBegin)
+        , endPtr(aEnd)
+    {
+    }
+
+    Span(Pointer aBegin, SizeType aCount)
+        : beginPtr(aBegin)
+        , endPtr(aBegin + aCount)
+    {
+    }
+
+    constexpr Reference operator[](SizeType aIndex)
+    {
+        assert(aIndex < Size());
+        return Data()[aIndex];
+    }
+
+    constexpr ConstReference operator[](SizeType aIndex) const
+    {
+        assert(aIndex < Size());
+        return Data()[aIndex];
+    }
+
+    [[nodiscard]] constexpr Reference At(SizeType aIndex)
+    {
+        if (aIndex >= Size())
+            throw std::out_of_range("Span::At: Position out of range");
+
+        return Data()[aIndex];
+    }
+
+    [[nodiscard]] constexpr ConstReference At(SizeType aIndex) const
+    {
+        if (aIndex >= Size())
+            throw std::out_of_range("Span::At: Position out of range");
+
+        return Data()[aIndex];
+    }
+
+    [[nodiscard]] constexpr Iterator Find(ConstReference aValue) noexcept
+    {
+        return Iterator(std::find(Begin(), End(), aValue));
+    }
+
+    [[nodiscard]] constexpr ConstIterator Find(ConstReference aValue) const noexcept
+    {
+        return ConstIterator(std::find(Begin(), End(), aValue));
+    }
+
+    [[nodiscard]] constexpr bool Contains(ConstReference aValue) const noexcept
+    {
+        return Find(aValue) != End();
+    }
+
+    [[nodiscard]] bool Contains(ConstIterator aPos) const noexcept
+    {
+        return Begin() <= aPos && aPos <= End();
+    }
+
+    [[nodiscard]] bool Contains(ConstIterator aFirst, ConstIterator aLast) const noexcept
+    {
+        return Begin() <= (std::min)(aFirst, aLast) && (std::max)(aLast, aFirst) <= End();
+    }
```

Comment:
When I think about `Contains`, I  would assume it is that the Span contains the value, not the iterator. I would say this is `IsInRange`. 

Besides this, is this function really necessary?